src/components/inputs/checkbox-input/checkbox-input.js

```diff
@@ -52,40 +63,38 @@ class CheckboxInput extends React.PureComponent {
 
   render() {
     return (
-      <div>
-        <label htmlFor={this.state.id} css={getLabelStyles(this.props)}>
-          <Spacings.Inline alignItems="center">
-            <div css={getCheckboxWrapperStyles(this.props)}>
-              {(() => {
-                if (this.props.isIndeterminate) return <Icons.Indeterminate />;
-                if (this.props.isChecked) return <Icons.Checked />;
-                return <Icons.Unchecked />;
-              })()}
-            </div>
-            {this.props.children && (
-              <Text.Body
-                // FIXME: add proper tones when we have disabled/primary in tones
-                tone={this.props.isDisabled ? 'secondary' : undefined}
-              >
-                {this.props.children}
-              </Text.Body>
-            )}
-            <input
-              css={css`
-                display: none;
-              `}
-              type="checkbox"
-              id={this.state.id}
-              name={this.props.name}
-              value={this.props.value}
-              onChange={this.props.onChange}
-              disabled={this.props.isDisabled}
-              checked={this.props.isChecked && !this.props.isIndeterminate}
-              {...filterDataAttributes(this.props)}
-            />
-          </Spacings.Inline>
-        </label>
-      </div>
+      <Label htmlFor={this.state.id} hasError={this.props.hasError}>
+        <Spacings.Inline alignItems="center">
+          <Checkbox
+            type="checkbox"
+            id={this.state.id}
+            name={this.props.name}
+            value={this.props.value}
+            onChange={this.props.onChange}
+            disabled={this.props.isDisabled}
+            checked={this.props.isChecked && !this.props.isIndeterminate}
+            isIndeterminate={this.props.isIndeterminate}
+            hasError={this.props.hasError}
+            {...filterDataAttributes(this.props)}
+          />
+
+          <div css={getCheckboxWrapperStyles(this.props)}>
+            {(() => {
```

Comment:
why not `do` expression?

Reply:
We decided not to use `do` expressions in UI Kit at an earlier date. There's probably a discussion about it somewhere.

Reply:
We don't use those features for uikit and appkit

Reply:
I would like to remove the use of our Icons here and use PURE css. But it's a bit more work. So maybe a followup.

Reply:
> There's probably a discussion about it somewhere.

oh my bad. carry on :)